src/hdmf_zarr/backend.py

```diff
@@ -1132,91 +1149,45 @@ def __read_dataset(self, zarr_obj, name):
         if dtype == 'scalar':
             data = zarr_obj[0]
 
-        obj_refs = False
-        reg_refs = False
-        has_reference = False
         if isinstance(dtype, list):
-            # compound data type
-            obj_refs = list()
-            reg_refs = list()
+            # Check compound dataset where one of the subsets contains references
+            has_reference = False
             for i, dts in enumerate(dtype):
-                if dts['dtype'] == DatasetBuilder.OBJECT_REF_TYPE:
-                    obj_refs.append(i)
+                if dts['dtype'] == 'object': # check items for object reference
```

Comment:
```suggestion
                if dts['dtype'] in ['object', 'region']: # check items for object reference
```

Reply:
For compound datasets, `BuilderZarrTableDataset` will need to handle the region references, so I think we can hear prepare the code to already wrap the dataset correctly so that if/when we add region reference supprort we will only need to update `BuilderZarrTableDataset`